content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,580 @@
+        return productCart.size();
+    }
+    public void removeFromCart(Product product){
+        productCart.remove(product);
+    }
+    public BigDecimal calculatePriceInCart(){
+        BigDecimal total=BigDecimal.ZERO;
+        for(Product product:productCart){
+       total= total.add(product.getCost());
+        }
+   return total;
+    }
+    public List<Product> displayAllItems(){
+       return productCart;
+    }
+}
+```
+### Controller Package
+The last package to review is the controller package. In the controller package we have our endpoints or APIs that connect to the internet all the basic functionalities of our services:
+```java
+@RestController
+@RequestMapping("cart")
+public class controller {
+    @Autowired
+    DataServiceImpl dataService;
+    @Autowired
+    CartServiceImpl cartService;
+   @PostMapping("/create_customer")
+   public Customer registerCustomer(@RequestBody Customer customer){
+ return  dataService.registerCustomer(customer);
+    }
+    @GetMapping("/get_all_customers")
+    public List<Customer> gettingAllList(){
+       return dataService.gettingAllCustomers();
+    }
+
+    @PostMapping("/create_admin")
+    public Admin registerAdmin(@RequestBody Admin admin){
+        return  dataService.registerAdmin(admin);
+    }
+    @GetMapping("/get_all_admins")
+    public List<Admin> getAllAdmin(){
+        return dataService.getAllAdmin();
+    }
+    @PostMapping("/{id}/add_product")
+    public void addProducts(@PathVariable Long id ,@RequestBody Product product){
+         cartService.addProductToCartByCustomer(id,product);
+    }
+@GetMapping("/cal_price")
+    public BigDecimal calculatingPrice(){
+       return cartService.calculatePrice();
+}
+@GetMapping("/{id}/payment")
+    public BigDecimal payment(@PathVariable Long id){
+       return cartService.calculatingPurchase(id);
+}
+@GetMapping("/{id}/view_all_product")
+    public List<Product> viewAllProducts(@PathVariable Long id){
+       return cartService.checkAllTheProductsInTheCartByCustomer(id);
+}
+@GetMapping("/{id}/view_all_productAdmin")
+    public List<Product> viewAllProduct(@PathVariable Long id){
+    return cartService.checkAllTheProductsInTheCartByAdmin(id);
+    }
+}
+```
+N/B `@PostMapping` is for saving objects, `@GetMapping` is for retrieving data from the database, `@PatchMapping` and `@PutMapping` is for updating data, and `@DeleteMapping` is for deleting. Also, note that we annotate the fields of this class with `@Autowire` to inject its dependencies to avoid initializing new objects. This is done by using these same objects each time they're needed throughout the whole codebase.
+### Running the server
+```java
+@SpringBootApplication
+public class ShoppingApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(ShoppingApplication.class, args);
+    }
+
+}
+```
+You have to click on the play button on the SpringBootApplication to run your server.
```

Comment:
Technically, that would only be for those using IDEs or particularly IntelliJ (i.e. other IDEs might have a different interface to run the application). Since you had in your prerequisites that the reader should have an intermediate knowledge of Java and a basic understanding of Spring Boot, I would even remove this section about running the server since you can safely assume they know how to.